### ISSUE TITLE
Unable to read metadata

### ISSUE THREAD

**Issue by vredesbyyrd** [desbma/sacad#30]
Hi, thanks for sharing this great tool. 

`Sacad_r` is returning `Unable to read metadata` on all directories I have tried running it on. 

Example cmd: 

```
Eluvium » sacad_r -d . 1000 cover.jpg                                      
sacad_r: Unable to read metadata for album directory './2003 - Lambent Material'      
sacad_r: Unable to read metadata for album directory './2004 - An Accidental Memory in the Case of Death'
sacad_r: Unable to read metadata for album directory './2005 - Talk Amongst the Trees'     
sacad_r: Unable to read metadata for album directory './2005 - Travels in Constants, Volume 20_ Behind Your Trouble'
sacad_r: Unable to read metadata for album directory './2006 - When I Live by the Garden and the Sea'
sacad_r: Unable to read metadata for album directory './2007 - Copia'                      
sacad_r: Unable to read metadata for album directory './2007 - Indecipherable Text/Disc 01'
sacad_r: Unable to read metadata for album directory './2007 - Indecipherable Text/Disc 02'
sacad_r: Unable to read metadata for album directory './2007 - Jesu _ Eluvium'              
sacad_r: Unable to read metadata for album directory './2014 - Pedals _ Petals'              
sacad_r: Unable to read metadata for album directory './2016 - Unity Engrossed Forever Supposedly'
sacad_r: Unable to read metadata for album directory './2017 - Shuffle Drones'                
sacad_r: Unable to read metadata for album directory './2019 - Pianoworks'                    
Analyzing library: 18dir [00:04,  2.78dir/s, files=160, albums=2, missing covers=1, errors=13]
sacad_r: Unable to find cover for 'Nightmare Ending' by 'Eluvium' from './2013 - Nightmare Ending/Disc 02/08 - Chime.mp3', './2013 - Nightmare Ending/Disc 02/09 - Rain Gently.mp3', './2013 - Nightmare Ending/Disc 02/10 - Impromptu (For the Procession).mp3', './2013 - Nightmare Ending/Disc 02/11 - Covered in Writing.mp3', './2013 - Nightmare Ending/Disc 02/12 - Entendre.mp3', './2013 - Nightmare Ending/Disc 02/13 - Strange Arrivals.mp3', './2013 - Nightmare Ending/Disc 02/14 - Happiness.mp3'
Searching covers: 100%|███████████| 1/1 [00:00<00:00,  2.81cover/s, ok=0, errors=0, no result found=1]
```
I use beets to tag my music, here is the metadata from _lambent material_ first track: 

```
Eluvium » beet info '/mnt/Data/Music/Albums/Eluvium/2003 - Lambent Material/01 - The Unfinished.flac'
/mnt/Data/Music/Albums/Eluvium/2003 - Lambent Material/01 - The Unfinished.flac
             album: Lambent Material
       albumartist: Eluvium
albumartist_credit: Eluvium
     albumdisambig: 
       albumstatus: Official
         albumtype: album
          arranger: 
               art: False
            artist: Eluvium
     artist_credit: Eluvium
              asin: B000093NQ4
          bitdepth: 16
           bitrate: 628101
        catalognum: TRR57
          channels: 2
          comments: 
              comp: False
          composer: 
           country: US
              date: 2003-01-01
              disc: 1
         disctitle: 
         disctotal: 1
            format: FLAC
             genre: Ambient, Electronic, Rock
            genres: Ambient, Electronic, Rock
          grouping: 
             label: Temporary Residence Limited
          language: eng
            length: 278.17333333333335
  mb_albumartistid: 86139012-695b-42f4-98f1-1f353f7230b4
        mb_albumid: 0a37bee7-e8bc-4902-bcd9-b99efd7b5d58
       mb_artistid: 86139012-695b-42f4-98f1-1f353f7230b4
 mb_releasegroupid: 58d3617c-6c24-3197-964f-3debcbc151ff
 mb_releasetrackid: 
        mb_trackid: c7d7e762-b4b6-4221-9989-cc3a31390192
             media: CD
     original_date: 2003-05-06
      original_day: 6
    original_month: 5
     original_year: 2003
   r128_album_gain: 0
   r128_track_gain: 0
        samplerate: 44100
            script: Latn
             title: The Unfinished
             track: 1
        tracktotal: 5
              year: 2003
```
Issue #23 appears to be similar to mine, but I have not been able to resolve it. I run arch linux, with `sacad 2.2.2-1` from the aur. Arch recently updated to python 3.8. Doubt that is the cause but figured it worth mentioning. 

Thanks for your time.

**Comment by desbma** [comment]
Hi, 

Please provide the output of `mutagen-inspect path_to_an_audio_file`.

If you are on Arch, the `python-mutagen` package is a dependency for Sacad, so you already have it installed.

**Comment by vredesbyyrd** [comment]
```
-- /mnt/Data/Music/Albums/Eluvium/2003 - Lambent Material/01 - The Unfinished.flac
- FLAC, 278.17 seconds, 44100 Hz (audio/flac)
ALBUM=Lambent Material
ALBUM ARTIST=Eluvium
ALBUMARTIST=Eluvium
ALBUMARTIST_CREDIT=Eluvium
MUSICBRAINZ_ALBUMCOMMENT=
MUSICBRAINZ_ALBUMSTATUS=Official
MUSICBRAINZ_ALBUMTYPE=album
ARRANGER=
ARTIST=Eluvium
ARTIST_CREDIT=Eluvium
ASIN=B000093NQ4
CATALOGNUMBER=TRR57
DESCRIPTION=
COMMENT=
COMPILATION=0
COMPOSER=
RELEASECOUNTRY=US
DATE=2003
YEAR=2003
DISC=1
DISCNUMBER=1
DISCSUBTITLE=
DISCTOTAL=1
DISCC=1
TOTALDISCS=1
GENRE=Ambient, Electronic, Rock
GROUPING=
LABEL=Temporary Residence Limited
PUBLISHER=Temporary Residence Limited
LANGUAGE=eng
MUSICBRAINZ_ALBUMARTISTID=86139012-695b-42f4-98f1-1f353f7230b4
MUSICBRAINZ_ALBUMID=0a37bee7-e8bc-4902-bcd9-b99efd7b5d58
MUSICBRAINZ_ARTISTID=86139012-695b-42f4-98f1-1f353f7230b4
MUSICBRAINZ_RELEASEGROUPID=58d3617c-6c24-3197-964f-3debcbc151ff
MUSICBRAINZ_RELEASETRACKID=
MUSICBRAINZ_TRACKID=c7d7e762-b4b6-4221-9989-cc3a31390192
MEDIA=CD
ORIGINALDATE=2003-05-06
R128_ALBUM_GAIN=0
R128_TRACK_GAIN=0
SCRIPT=Latn
TITLE=The Unfinished
TRACK=1
TRACKNUMBER=1
TRACKTOTAL=5
TRACKC=5
TOTALTRACKS=5
```

**Comment by desbma** [comment]
Can you run:
`pacman -Qqo /usr/lib/python3.7`

The Arch Python 3.8 upgrade broke every AUR package built on Python 3.7, so maybe a dependency is broken ?

If nothing, or nothing related to Sacad is shown, you can upload a track, or the beginning of it (truncated with for example `dd if=INPUT_FILE of=OUTPUT_FILE bs=128K count=1`), so I can look at it.

**Comment by vredesbyyrd** [comment]
Thanks for the tips. When I did the python 3.8 update I ran `pacman -Qqo /usr/lib/python3.7` and rebuilt the few packages that were still under 3.7, so I do not think it is a dependency issue :( That is what I was initially leaning towards though.

Since I use [beets](https://github.com/beetbox/beets) to tag my music, I also use its [fetchart](https://beets.readthedocs.io/en/stable/plugins/fetchart.html) plugin to pull in album covers. Fetchart was unable to find about 150 covers so I wanted to see if there were any other tools out there before I manually searched for them. That brought me to `sacad` :) 

The non recursive `sacad` works fine for me, and I already tackled about half of the missing covers that way, so I am personally not concerned about it. But it still begs the question on why `sacad_r` is having trouble on my system. I am 90% percent sure there is nothing funky going on with my python environment. 

If you want to take a look, here is a link to a truncated flac file, on my onedrive account. [sample flac](https://1drv.ms/u/s!Au0J8kZrcsZnjL9re7Ml6lVqtmdwWg?e=MiJXhV)
